### PR TITLE
[Fix] Permanently delete `ERROR` and `TERMINATED` state clusters if their creation fails

### PR DESCRIPTION
**Pull request by tanmay-db**
## Changes

If we get error or terminated cluster (after getting WaitGetClusterRunning in Create) then we permanently delete them. 

## Tests

Unit tests
- [ ] `make test` run locally
- [ ] relevant change in `docs/` folder
- [ ] covered with integration tests in `internal/acceptance`
- [ ] relevant acceptance tests are passing
- [ ] using Go SDK

### CODE REVIEW

File: clusters/resource_cluster.go [hunk 1]
```diff
@@ -472,6 +472,8 @@ func resourceClusterCreate(ctx context.Context, d *schema.ResourceData, c *commo
 
 	clusterInfo, err := clusterWaiter.GetWithTimeout(timeout)
 	if err != nil {
+		// In case of "ERROR" or "TERMINATED" state, WaitGetClusterRunning returns an error and we should delete the cluster before returning
+		resourceClusterDelete(ctx, d, c)
```
Comment (alexott) [inline clusters/resource_cluster.go:476]: we need to show the error if we can't delete the resource, etc.

Comment (alexott) [reply clusters/resource_cluster.go:476]: Like this: https://github.com/databricks/terraform-provider-databricks/blob/main/vectorsearch/resource_vector_search_endpoint.go#L53

**Review by alexott — CHANGES_REQUESTED** [review]
We should expose nested error...

File: clusters/resource_cluster.go (hunk 1 above)
Comment (tanmay-db) [reply clusters/resource_cluster.go:476]: Yes, I was just testing a few things with directly calling deletion method before sending it in review, will update it.

**Review by alexott — APPROVED** [review]

**Pull request merged**